File: app/src/main/java/com/wojdor/hearthstonecards/data/database/CardDatabase.java
package com.wojdor.hearthstonecards.data.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.wojdor.hearthstonecards.data.database.dao.CardDao;
import com.wojdor.hearthstonecards.domain.Card;

@Database(entities = {Card.class}, version = 1, exportSchema = false)
public abstract class CardDatabase extends RoomDatabase {

    private static final String DATABASE_NAME = "card_database";
    private static final Object LOCK = new Object();
    private static CardDatabase instance;

    public static CardDatabase getInstance(Context context) {
        if (instance == null) {
            synchronized (LOCK) {
                instance = Room.databaseBuilder(context.getApplicationContext(),
                        CardDatabase.class, DATABASE_NAME)
                        .build();
            }
        }
        return instance;
    }

    public abstract CardDao cardDao();
}

File: app/src/main/java/com/wojdor/hearthstonecards/data/database/CardDatabase.kt
package com.wojdor.hearthstonecards.data.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.wojdor.hearthstonecards.domain.Card

@Database(entities = [(Card::class)], version = 1, exportSchema = false)
abstract class CardDatabase : RoomDatabase() {

    abstract fun cardDao(): CardDao

    companion object {
        const val DATABASE_NAME = "card_db"
    }
}
